compiler/rustc_middle/src/ty/consts/kind.rs

```diff
@@ -29,23 +28,15 @@ impl rustc_errors::IntoDiagnosticArg for Unevaluated<'_> {
 
 impl<'tcx> Unevaluated<'tcx> {
     #[inline]
-    pub fn shrink(self) -> Unevaluated<'tcx, ()> {
-        debug_assert_eq!(self.promoted, None);
-        Unevaluated { def: self.def, substs: self.substs, promoted: () }
+    pub fn expand(self) -> mir::Unevaluated<'tcx> {
```

Comment:
i would love for us to not have that method, instead changing shared methods to again use def + substs separately :thinking: afaict that method is currently unused?

Reply:
well, i don't think we should ever convert `ty::Unevaluated` to `mir::Unevaluated` ^^ didn't see that this method is used, going to look at this at a later point then